### PR TITLE
Options

### LINKED ISSUE #160: Show users what's new in the current version
The ideal version wouldn't take users away from LibraryThing, but the simple version would just open GitHub in a new tab

### PR DESCRIPTION
**Pull request by braxtonhall**
toward #141 
resolves #160 

I was thinking about it, and maybe both of these issues are kinda dumb.

## RE #141 (Options)
There's no reason to let users change the frequency. If it's too high, they hit the API limit. If it's too low, then they miss important updates. Most importantly, from the user's point of view, when they get the notification is when the new version is out -- they don't need it the second it's out.

I am still open to options for form size settings, but it's also a bit awkward to describe. At the moment, it's a QOL thing that I don't think anyone actually even notices.

## RE #160 (What's new in the latest version)
I don't think your average user cares about what's new in the latest version. If anything, I think it might confuse people, because the "blue toast" means new download to some of the catalogers already.

##

I still want to merge this, as I think this is a step forward for the codebase, but I don't know if I want this in 1.3.0

### CODE REVIEW

**Comment by braxtonhall** [comment]
Although MAYBE, we could just make the "Looks like you're on the new version" toast a `SUCCESS` toast instead of an `INFO` toast? And then why not have the notif?

File: extension/ts/content/extensions/version/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	get: (): Promise<Version> => storage.get<Version>(VERSION_SEEN_KEY, {major: 0, minor: 0, revision: 0}),
+	set: (): Promise<Version> => storage.set(VERSION_SEEN_KEY, getCurrentVersion()),
+};
+
+const checkedTime = {
+	get: (): Promise<number> => storage.get(LAST_CHECKED_KEY, 0),
+	set: (): Promise<number> => storage.set(LAST_CHECKED_KEY, Date.now()),
+};
+
+const beenAWhile = async (): Promise<boolean> => Date.now() > (await checkedTime.get()) + (await getCheckInterval());
```
Comment (adirar111) [inline extension/ts/content/extensions/version/index.ts:23]: lmao

File: extension/ts/content/extensions/version/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+				showToast("New version available!\n\nClick here to download it.", ToastType.INFO, () =>
+					window.open(download)
+				);
+		} else if (versionEquals(remoteVersion, currentVersion)) {
+			// html &&
+			// 	showToast(
+			// 		"Looks like this is your first time on this version of Better LibraryThing!\n\nClick here to see what's new.",
+			// 		ToastType.INFO,
+			// 		() => window.open(html)
+			// 	);
```
Comment (adirar111) [inline extension/ts/content/extensions/version/index.ts:42]: did you wanna scrap this?

File: extension/ts/content/extensions/version/version.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	} else if (comparator(versionB.minor, versionA.minor)) {
+		return false;
+	} else if (comparator(versionA.revision, versionB.revision)) {
+		return true;
+	} else if (comparator(versionB.revision, versionA.revision)) {
+		return false;
+	} else {
+		return false;
+	}
+};
```
Comment (adirar111) [inline extension/ts/content/extensions/version/version.ts:23]: i don't have any issues with this, and i'm sure it's good for readability but it kinda reminds me of this lol

https://twitter.com/ctrlshifti/status/1288745146759000064

**Review by adirar111 — APPROVED** [review]
am good with merging this even if you think the original motivating tickets might need more consideration

**Comment by adirar111** [comment]
> Although MAYBE, we could just make the "Looks like you're on the new version" toast a `SUCCESS` toast instead of an `INFO` toast? And then why not have the notif?

this seems reasonable. also could be an entry point for them to update their internal docs with the new features in the release (?)

File: extension/ts/content/extensions/version/version.ts (hunk 3 above)
Comment (braxtonhall) [reply extension/ts/content/extensions/version/version.ts:23]: lol this is what it used to look like
https://github.com/braxtonhall/library-thing/blob/b482ff6554b14a9334a7ecbb0330e55883ab0d01/extension/ts/content/extensions/version.ts#L45-L58

there _must_ be a better way

File: extension/ts/content/extensions/version/index.ts (hunk 2 above)
Comment (braxtonhall) [reply extension/ts/content/extensions/version/index.ts]: restored

**Pull request merged**